src/file.cr

```diff
@@ -3,10 +3,9 @@ require "c/stdio"
 require "c/stdlib"
 require "c/sys/stat"
 require "c/unistd"
-require "path"
```

Comment:
Removing the require and then using it to define `SEPARATOR` from `Path::SEPARATOR` is not what I meant with my previous comment.

You're adding `Path` as dependency of `File`, not just by the require, but also all the methods owned by File now depend on an instance of `Path` that is discarded when used.